src/io/flutter/perf/EditorPerfDecorations.java

```diff
@@ -0,0 +1,502 @@
+   * Returns the action executed when the icon is left-clicked.
+   *
+   * @return the action instance, or null if no action is required.
+   */
+  @Nullable
+  public AnAction getClickAction() {
+    return new AnAction() {
+      @Override
+      public void actionPerformed(AnActionEvent e) {
+        if (isActive()) {
+
+          final ToolWindowManagerEx toolWindowManager = ToolWindowManagerEx.getInstanceEx(getApp().getProject());
+          final ToolWindow flutterToolWindow = toolWindowManager.getToolWindow(FlutterView.TOOL_WINDOW_ID);
+          if (flutterToolWindow.isVisible()) {
+            showPerfViewMessage();
+            return;
+          }
+          flutterToolWindow.show(() -> showPerfViewMessage());
+        }
+      }
+    };
+  }
+
+  private void showPerfViewMessage() {
+    final FlutterView flutterView = ServiceManager.getService(getApp().getProject(), FlutterView.class);
+    final InspectorPerfTab inspectorPerfTab = flutterView.showPerfTab(getApp());
+    final StringBuilder sb = new StringBuilder("<html><body>");
+    final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss");
+    sb.append("<h2>Widget rebuild stats for ");
+    sb.append(Objects.requireNonNull(perfModelForFile.getTextEditor().getFile()).getName());
+    sb.append(" at ");
+    sb.append(formatter.format(LocalDateTime.now()));
+    sb.append("</h2>");
+    for (String line : getTooltipLines()) {
+      sb.append("<h3>");
+      sb.append(line);
+      sb.append("</h3>");
+    }
+
+    sb.append("<p>");
+    sb.append("Rebuilding widgets is generally very cheap. You should only worry " +
+              "about optimizing code to reduce the the number of widget rebuilds " +
+              "if you notice that the frame rate is bellow 60fps or if widgets " +
+              "that you did not expect to be rebuilt are rebuilt a very large " +
+              "number of times.");
+    sb.append("</p>");
+    sb.append("</body></html>");
+    inspectorPerfTab.getWidgetPerfPanel().setPerfMessage(perfModelForFile.getTextEditor(), range, sb.toString());
+  }
+
+  @NotNull
+  @Override
+  public Alignment getAlignment() {
+    return Alignment.LEFT;
+  }
+
+  @NotNull
+  @Override
+  public Icon getIcon() {
+    lastIcon = getIconInternal();
+    return lastIcon;
+  }
+
+  public Icon getIconInternal() {
+    final int count = getCountPastSecond();
+    if (count == 0) {
+      return perfModelForFile.isHoveredOverLineMarkerArea() ? FlutterIcons.CustomInfo : EMPTY_ICON;
+    }
+    if (count > HIGH_LOAD_THRESHOLD) {
+      return RED_PROGRESS;
+    }
+    return NORMAL_PROGRESS;
+  }
+
+  Color getErrorStripeMarkColor() {
+    // TODO(jacobr): tween from green or blue to red depending on the count.
+    final int count = getCountPastSecond();
+    if (count == 0) {
+      return null;
+    }
+    if (count > HIGH_LOAD_THRESHOLD) {
+      return JBColor.RED;
+    }
+    return JBColor.YELLOW; // TODO(jacobr): should we use green here instead?
```

Comment:
Did you settle on a color here? I assume below the threshold, the color is there to indicate that we're repaiting the widget, but likely not at a rate which is an issue?

Reply:
I'm leaning towards yellow as yellow as a background color just seems like a good low intensity highlight but I could be convinced otherwise. It would be nice to settle on a good grey->green->yellow->red scheme for all performance related stats.